src/tests/match-requests.spec.ts

```diff
@@ -115,7 +118,54 @@ describe("Match requests", () => {
         key3: "c"
       },
       field2: "d",
-      field3: 1
+      field3: 1,
+      requestCount: 2,
+    });
+  });
+
+  describe('When more than one requests are the same', () => {
+    let response2;
+
+    beforeAll(async () => {
+      await axios.post(`${PROXAY_HOST}/__proxay/tape`, {
+        tape: "tape",
+        mode: "record"
+      });
+
+      await axios.post(`${PROXAY_HOST}${JSON_IDENTITY_PATH}`, {
+        field3: 1,
+      });
+
+      await axios.post(`${PROXAY_HOST}${JSON_IDENTITY_PATH}`, {
+        field3: 1,
+      });
     });
+
+    it("Picks the request that hasn't been matched over the request has been matched", async () => {
+      await axios.post(`${PROXAY_HOST}/__proxay/tape`, {
+        tape: "tape",
+        mode: "replay"
+      });
+
+      const response1 = await axios.post(`${PROXAY_HOST}${JSON_IDENTITY_PATH}`, {
+        field3: 1
+      });
+
+      response2 = await axios.post(`${PROXAY_HOST}${JSON_IDENTITY_PATH}`, {
+        field3: 1
+      });
+
+      expect(response1.data.requestCount).not.toEqual(response2.data.requestCount);
+    });
+
+    describe('when there are more requests than recorded', () => {
+      it("Pick the last request", async () => {
+        const response3 = await axios.post(`${PROXAY_HOST}${JSON_IDENTITY_PATH}`, {
```

Comment:
I'd prefer if this test was independent from other tests, even if that means copy-pasting `response1` and `response2` (otherwise this will fail if you run tests separately). I'd remove the shared `let response2;` too.

Reply:
Done